src/watertap_contrib/reflo/unit_models/thermal_energy_storage.py

```diff
@@ -0,0 +1,454 @@
+        ),
+    )
+
+    def build(self):
+        super().build()
+
+        # Add hx inlet block
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = True
+        self.hx_inlet_block = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of exit stream from the heat source",
+            **tmp_dict,
+        )
+
+        # Add hx outlet block
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = False  # block is not an inlet
+        self.hx_outlet_block = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of stream exiting the TES back to the heat source",
+            **tmp_dict,
+        )
+
+        # Add process inlet block
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = True
+        self.process_inlet_block = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of exit stream from the heat source",
+            **tmp_dict,
+        )
+
+        # Add process outlet block
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = False  # block is not an inlet
+        self.process_outlet_block = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of stream exiting the TES back to the heat source",
+            **tmp_dict,
+        )
+
+        # Add Ports
+        self.add_inlet_port(name="tes_hx_inlet", block=self.hx_inlet_block)
+        self.add_outlet_port(name="tes_hx_outlet", block=self.hx_outlet_block)
+        self.add_inlet_port(name="tes_process_inlet", block=self.process_inlet_block)
+        self.add_outlet_port(name="tes_process_outlet", block=self.process_outlet_block)
+
+        # From https://github.com/gmlc-dispatches/dispatches/blob/main/dispatches/properties/solarsalt_properties.py
+        # **** Requires Temperature in K
+        # Ref: (2015) Chang et al, Energy Procedia 69, 779 - 789
+        # Specific Heat Capacity as a function of Temperature, J/kg/K
+        # def specific_heat_cp(model):
+        #    return model.specific_heat_cp
+        #       == 1443 + 0.172 * model.temperature
+
+        # TODO: Add multiple tanks?
```

Comment:
```suggestion
        # TODO: Add multiple tanks?
```
I would do that in a flowsheet rather than pack them in here (maybe?)-unless you could account for one lumped tank to account for multiple.